File: samples/java/TicTacToe/Server/src/main/java/systems/zlink/samples/tictactoe/server/configuration/SampleLogging.java
package systems.zlink.samples.tictactoe.server.configuration;

import java.io.IOException;
import java.nio.file.Files;
import java.nio.file.Path;

public final class SampleLogging {
    private SampleLogging() {}

    public static void configure(SampleLogSettings settings, String role) {
        try {
            Files.createDirectories(Path.of(settings.logDirectory()));
            Path.of(settings.logDirectory(), role + ".log").toFile().createNewFile();
            Files.createDirectories(Path.of(settings.logDirectory()));
        } catch (IOException ex) {
            throw new IllegalStateException("Failed to initialize sample log files.", ex);
        }
    }
}

File: samples/kotlin/TicTacToe/Server/src/main/kotlin/systems/zlink/samples/kotlin/tictactoe/server/configuration/SampleLogging.kt
package systems.zlink.samples.kotlin.tictactoe.server.configuration

import java.nio.file.Files
import java.nio.file.Path

object SampleLogging {
    fun configure(settings: SampleSettings, role: String) {
        Files.createDirectories(Path.of(settings.logDirectory))
        Path.of(settings.logDirectory, "$role.log").toFile().createNewFile()
        Files.createDirectories(Path.of(settings.logDirectory))
    }
}
